Document/0x06b-Basic-Security-Testing.md

```diff
@@ -176,13 +163,60 @@ $ docker run -it -p 8000:8000 opensecurity/mobile-security-framework-mobsf:lates
 
 ##### Remote Shell
 
-##### Connecting to a physical iOS device
+In contrast to Android where you can easily access the device shell using the adb tool, on iOS you only have the option to access the remote shell via SSH. This means also that your iOS device must be jailbroken in order to connect to its shell from your host computer. For this section we assume that you've properly jailbroken your device and have either Cydia (see screenshot above) or Sileo installed as introduced in "Getting Privileged Access". In the rest of the guide we will reference to Cydia, but the same packages should be available in Sileo.
+
+<img src="Images/Chapters/0x06b/cydia.png" alt="iOS App Folder Structure" width="250">
+
+In order to enable SSH access to your iOS device you can install the OpenSSH package. Once installed, be sure to connect both devices to the same Wi-Fi network and take a note of the device IP address, which you can find in the Settings -> Wi-Fi menu and tapping once on the info icon of the network you're connected to.
+
+You can access now the remote device's shell by running `ssh root@<device_ip_address>` and you'll already have root access, for example:
+
+```shell
+$ ssh root@192.168.197.234
+root@192.168.197.234's password:
+iPhone:~ root#
+```
+
+> press Control + D or type `exit` to quit
+
+When accessing your iOS device via SSH consider the following:
 
-##### Connecting to an iOS emulator
+- The default users are `root` and `mobile`.
+- The default password is `alpine`.
+
+Remember to change the default password for both users `root` and `mobile` as anyone being in the same network and knowing the IP address of your device an the well-know default password will be able to access your device as root.
+
+###### Connect to a Device via SSH over USB
+
+During a real black box test, a reliable Wi-Fi connection may not be available. In this situation, you can use [usbmuxd](https://github.com/libimobiledevice/usbmuxd "usbmuxd") to connect to your device's SSH server via USB.
+
+Usbmuxd is a socket daemon that monitors USB iPhone connections. You can use it to map the mobile device's localhost listening sockets to TCP ports on your host machine. This allows you to conveniently SSH into your iOS device without setting up an actual network connection. When usbmuxd detects an iPhone running in normal mode, it connects to the phone and begins relaying requests that it receives via `/var/run/usbmuxd`.
+
+Connect macOS to an iOS device by installing and starting iproxy:
+
+```shell
+$ brew install libimobiledevice
+$ iproxy 2222 22
+waiting for connection
+```
+
+The above command maps port `22` on the iOS device to port `2222` on localhost. With the following command in a new terminal window, you can connect to the device:
+
+```shell
+$ ssh -p 2222 root@localhost
+root@localhost's password:
+iPhone:~ root#
+```
+
+You can also connect to your iPhone's USB via [Needle](https://labs.mwrinfosecurity.com/blog/needle-how-to/ "Needle").
```

Comment:
do we see the objection/frida repl as a sort of shell as well "inapp" ?

Reply:
not really, this section is only about a "pure" shell on the device. Needle was added because it also allows that with:

```
[needle] > shell
[*] Spawning a shell...
[*] Checking connection with device...
[V] Connection not present, creating a new instance
[V] Setting up USB port forwarding on port 2222
[V] Setting up SSH connection...
[+] Connected to: 127.0.0.1
Warning: Permanently added '[127.0.0.1]:2222' (RSA) to the list of known hosts.
MWR-iPhone:~ root# id
uid=0(root) gid=0(wheel) groups=0(wheel),...
```